zebra/zebra_router.h

```diff
@@ -71,6 +71,36 @@ struct zebra_mlag_info {
 
 	/* The system mac being used */
 	struct ethaddr mac;
+	/*
+	 * Zebra will open the communication channel with MLAGD only if any
+	 * clients are interested and it is controlled dynamically based on
+	 * client registers & un-registers.
+	 */
+	uint32_t clients_interested_cnt;
+
+	/* coomunication channel with MLAGD is established */
+	bool connected;
+
+	/* connection retry timer is running */
+	bool timer_running;
+
+	/* Holds the client data(unencoded) that need to be pushed to MCLAGD*/
+	struct stream_fifo *mlag_fifo;
+
+	/*
+	 * A new Kernel thread will be created to post the data to MCLAGD.
+	 * where as, read will be performed from the zebra main thread, because
+	 * read involves accessing client registartion data structures.
+	 */
+	struct frr_pthread *zebra_pth_mlag;
+
+	/* MLAG Thread context 'master' */
+	struct thread_master *th_master;
+
+	/* Threads for read/write. */
+	struct thread *t_read;
+	struct thread *t_write;
+	pthread_mutex_t mlag_th_mtx;
```

Comment:
so now there's still this mutex - but it seems it's never used. as I tried to say yesterday: what is being protected, from whom, in what circumstances?

Reply:
can i  add a comment, to indicate in case going forward if Zebra MLAG Thread need to access any Zebra Global data structures it should be accessed with proper locking, currently MLAG thread will not access any Zebra Globals except in thread apis.

Reply:
is that sufficient (or) can i remove this mutex..?

Reply:
it's clearer if there's no unused mutex sitting around, if it's not needed.

Reply:
ok, will delete it